awx/ui_next/src/components/CodeMirrorInput/VariablesDetail.jsx

```diff
@@ -14,9 +14,7 @@ function VariablesDetail({ value, label, rows }) {
   const [currentValue, setCurrentValue] = useState(value);
   const [error, setError] = useState(null);
 
-  if (!value) {
-    return null;
-  }
+  useEffect(() => setCurrentValue(value), [value]);
```

Comment:
We'll need to check the `mode` and then check the value to see if it's in the corresponding data format. If it isn't, we need to convert the value to the correct mode (similar to lines 42–45). Otherwise `mode` and `currentValue` could fall out of sync